app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,43 @@
+class ReservationsController < ApplicationController
+  def index
+    @user = User.where(id: params[:user_id]).first
+    @reservations = @user.reservations
+    @new_res = @reservations.map do |reservation|
+      reservation.as_json.merge(house: reservation.house_ids[0] || {})
+    end
+    render json: @new_res
+  end
+
+  def show
+    @reservation = Reservation.where(id: params[:id]).first
+    render json: @reservation
+  end
+
+  def create
+    @user = User.where(id: params[:user_id]).first
+    @reservation = @user.reservations.create(reservation_params)
+    @reservation.house_ids = params[:house_ids]
+
+    if @reservation.save
+      render json: { message: 'Created' }, status: :created
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @reservation = Reservation.where(id: params[:id]).first
```

Comment:
- [ ] Using `find` here should be a better option then `where`, since you already have the `id`. This way you won't even need to use `first`.